test/Microsoft.Data.Analysis.Tests/DataFrameTests.cs

```diff
@@ -1669,6 +1669,190 @@ public void TestMerge()
             VerifyMerge(merge, left, right, JoinAlgorithm.Inner);
         }
 
+        private void MatchRowsOnMergedDataFrame(DataFrame merge, DataFrame left, DataFrame right, long mergeRow, long? leftRow, long? rightRow)
+        {
+            Assert.Equal(merge.Columns.Count, left.Columns.Count + right.Columns.Count);
+            DataFrameRow dataFrameMergeRow = merge.Rows[mergeRow];
+            int columnIndex = 0;
+            foreach (object value in dataFrameMergeRow)
+            {
+                object compare = null;
+                if (columnIndex < left.Columns.Count)
+                {
+                    if (leftRow != null)
+                    {
+                        compare = left.Rows[leftRow.Value][columnIndex];
+                    }
+                }
+                else
+                {
+                    int rightColumnIndex = columnIndex - left.Columns.Count;
+                    if (rightRow != null)
+                    {
+                        compare = right.Rows[rightRow.Value][rightColumnIndex];
+                    }
+                }
+                Assert.Equal(value, compare);
+                columnIndex++;
+            }
+        }
+
+        [Theory]
+        [InlineData(10, 5, JoinAlgorithm.Left)]
+        [InlineData(5, 10, JoinAlgorithm.Right)]
+        public void TestMergeEdgeCases_Left(int leftLength, int rightLength, JoinAlgorithm joinAlgorithm)
+        {
+            DataFrame left = MakeDataFrameWithAllMutableColumnTypes(leftLength);
+            if (leftLength > 5)
+            {
+                left["Int"][8] = null;
+            }
+            DataFrame right = MakeDataFrameWithAllMutableColumnTypes(rightLength);
+            if (rightLength > 5)
+            {
+                right["Int"][8] = null;
+            }
+
+            DataFrame merge = left.Merge<int>(right, "Int", "Int", joinAlgorithm: joinAlgorithm);
+            Assert.Equal(10, merge.Rows.Count);
+            Assert.Equal(merge.Columns.Count, left.Columns.Count + right.Columns.Count);
+            int[] matchedFullRows = new int[] { 0, 1, 3, 4 };
+            for (long i = 0; i < matchedFullRows.Length; i++)
+            {
+                int rowIndex = matchedFullRows[i];
+                MatchRowsOnMergedDataFrame(merge, left, right, rowIndex, rowIndex, rowIndex);
```

Comment:
>MatchRowsOnMergedDataFrame [](http://example.com/codeflow?start=16&length=26)

Can you also add a test case for the scenario reported in the issue?

Reply:
It's actually a part of TestMergeEdgeCases_Left, but I added one now anyway